lib/absinthe/type/built_ins/introspection.ex

```diff
@@ -226,10 +242,19 @@ defmodule Absinthe.Type.BuiltIns.Introspection do
 
     field :args,
       type: non_null(list_of(non_null(:__inputvalue))),
-      resolve: fn _, %{source: %{args: args}} ->
+      args: [
+        include_deprecated: [
+          type: :boolean,
+          default_value: false
+        ]
+      ],
+      resolve: fn %{include_deprecated: show_deprecated}, %{source: %{args: args}} ->
         args =
           args
           |> Map.values()
+          |> Enum.filter(fn %{deprecation: is_deprecated} ->
+            !is_deprecated || show_deprecated
+          end)
```

Comment:
Sorry as one last revision: I'm not normally super DRY about things but this is the same function call 4 times over in the same module, can you extract this into a `defp filter_deprecated` function?

Reply:
Good idea 👍